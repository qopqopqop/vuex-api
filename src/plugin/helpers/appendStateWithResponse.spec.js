import mutateStateWithResponse from './appendStateWithResponse'

const set = (state, key, obj) => {
  state[key] = obj;
  return state
}



describe('Api handler', () => {
  it('should throw error when no keyPath', () => {
    const state =  {};
    const resp = { status: 'loading', data: { toto: 'tutu' } }
    expect(() => mutateStateWithResponse(state, undefined, resp)).toThrow()
  })

  it('should mutate correctly on keyPath with one item', () => {
    const state =  {};
    const resp = { status: 'loading', data: { toto: 'tutu' } }
    const keyPath = ['foo']
    const { stateObject, head } = mutateStateWithResponse(state, keyPath, resp)
    expect(set(state, head, stateObject )).toMatchSnapshot()
  })

  it('should mutate correctly on keyPath with more than one item', () => {
    const state =  {};
    const resp = { status: 'loading', data: { toto: 'tutu' } }
    const keyPath = ['foo', 'bar', 'baz'];
    const { stateObject, head } = mutateStateWithResponse(state, keyPath, resp)
    expect(set(state, head, stateObject )).toMatchSnapshot()
  })

  it('should mutate correctly already existing key on keyPath with one item', () => {
    const state =  { foo: { status: 'success', data: 'bax' } };
    const resp = { status: 'loading', data: { toto: 'tutu' } }
    const keyPath = ['foo']
    const { stateObject, head } = mutateStateWithResponse(state, keyPath, resp)
    expect(set(state, head, stateObject )).toMatchSnapshot()
  })

  it('should mutate correctly already existing key on keyPath with more than one item', () => {
    const state =  { foo: { status: 'success', data: 'bax' } };
    const resp = { status: 'loading', data: { toto: 'tutu' } }
    const keyPath = ['foo', 'bar', 'baz'];
    const { stateObject, head } = mutateStateWithResponse(state, keyPath, resp)
    expect(set(state, head, stateObject )).toMatchSnapshot()
  })
})
