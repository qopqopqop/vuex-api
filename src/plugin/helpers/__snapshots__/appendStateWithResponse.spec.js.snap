// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Api handler should mutate correctly already existing key on keyPath with more than one item 1`] = `
Object {
  "foo": Object {
    "bar": Object {
      "baz": Object {
        "data": Object {
          "toto": "tutu",
        },
        "status": "loading",
      },
    },
    "data": "bax",
    "status": "success",
  },
}
`;

exports[`Api handler should mutate correctly already existing key on keyPath with one item 1`] = `
Object {
  "foo": Object {
    "data": Object {
      "toto": "tutu",
    },
    "status": "loading",
  },
}
`;

exports[`Api handler should mutate correctly on keyPath with more than one item 1`] = `
Object {
  "foo": Object {
    "bar": Object {
      "baz": Object {
        "data": Object {
          "toto": "tutu",
        },
        "status": "loading",
      },
    },
  },
}
`;

exports[`Api handler should mutate correctly on keyPath with one item 1`] = `
Object {
  "foo": Object {
    "data": Object {
      "toto": "tutu",
    },
    "status": "loading",
  },
}
`;
